File: backend/src/main/java/com/offerflow/service/InterviewService.java
package com.offerflow.service;

import java.util.List;

import com.offerflow.dto.request.CreateInterviewRequest;
import com.offerflow.dto.request.UpdateInterviewRequest;
import com.offerflow.dto.response.InterviewResponse;
import com.offerflow.entity.User;

public interface InterviewService {

    InterviewResponse createInterview(
            CreateInterviewRequest request,
            User user);

    List<InterviewResponse> getInterviews(
            User user);

    InterviewResponse getInterviewById(
            Long id,
            User user);

    InterviewResponse updateInterview(
            Long id,
            UpdateInterviewRequest request,
            User user);

    void deleteInterview(
            Long id,
            User user);
}

File: frontend/src/services/interviewService.ts
import api from "@/api/axios";

import type { Interview } from "@/types/interview";
import type { InterviewRequest } from "@/types/interviewRequest";

export async function getInterviews() {
  const response = await api.get<Interview[]>("/interviews");

  return response.data;
}

export async function getInterviewById(
  id: number,
) {
  const response = await api.get<Interview>(
    `/interviews/${id}`,
  );

  return response.data;
}

export async function createInterview(
  request: InterviewRequest,
) {
  const response = await api.post<Interview>(
    "/interviews",
    request,
  );

  return response.data;
}

export async function updateInterview(
  id: number,
  request: InterviewRequest,
) {
  const response = await api.put<Interview>(
    `/interviews/${id}`,
    request,
  );

  return response.data;
}

export async function deleteInterview(
  id: number,
) {
  await api.delete(`/interviews/${id}`);
}